pkg/controller/projectreference/projectreference_controller.go

```diff
@@ -121,67 +122,58 @@ func (r *ReconcileProjectReference) Reconcile(request reconcile.Request) (reconc
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileProjectReference) ReconcileHandler(adapter *ReferenceAdapter, reqLogger logr.Logger) (reconcile.Result, error) {
-	// Set conditions
-	prState, err := adapter.EnsureProjectReferenceInitialized()
-	if prState == ObjectModified || err != nil {
-		return r.requeueOnErr(err)
-	}
 
-	// Cleanup
-	if adapter.IsDeletionRequested() {
-		err := adapter.EnsureProjectCleanedUp()
-		if err != nil {
-			return r.requeueAfter(5*time.Second, err)
-		}
-		return r.doNotRequeue()
-	}
+type OperationResult struct {
+	RequeueDelay   time.Duration
+	RequeueRequest bool
+	CancelRequest  bool
+}
 
-	// Make projectReference  be processed based on state of ProjectClaim and Project Reference
-	claimStatus, err := adapter.EnsureProjectClaimReady()
-	if claimStatus == gcpv1alpha1.ClaimStatusReady || err != nil {
-		return r.requeueOnErr(err)
-	}
+func StopProcessing() (OperationResult, error) {
+	return OperationResult{0 * time.Second, false, true}, nil
```

Comment:
does this still work?
```suggestion
	return OperationResult{0, false, true}, nil
```

Reply:
it does! Thanks